Microsoft.Toolkit.Uwp.UI.Controls/CoordinatedPanel/CoordinatablePanel.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.Toolkit.Uwp.UI.Extensions;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Merges this panel's layout process with the contents of an inner <see cref="ItemsPresenter"/> control's items. The parent <see cref="ItemsControl.ItemsPanel"/> must be set to an <see cref="InsetPanel"/> and this panels <see cref="IsCoordinating"/> property must be set to true. This scenario meant for use within the <see cref="ControlTemplate"/> of an <see cref="ItemsControl"/>.
+    /// </summary>
+    public abstract class CoordinatablePanel : Panel, ICoordinatingPanel
+    {
+        // TODO: Does it make sense to have multiple ItemsPresenters supported?
+        private InsetPanel _insetPanel;
+        private ItemsPresenter _coordinatedItemsPresenter;
+        private ItemsControl _itemsController;
+
+        /// <inheritdoc/>
+        public bool IsCoordinating { get; set; }
```

Comment:
Should this be a DP?

Reply:
hmm, not sure. It's not likely to be a bind target or to change.
I'll check with Michael and see what he thinks.